### PR TITLE
cpusupport fixes

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
Context for 10dad29 is #453.  That commit could have merely been "replaced `CFLAGS` with `CPPFLAGS`" instead of "use `CPPFLAGS` as well as `CFLAGS`", but I figure there's no harm in using both in `cpusupport.sh`.

File: cpusupport/cpusupport.h [hunk 1]
```diff
@@ -114,13 +114,21 @@
...
  * relevant macro may be defined (e.g., by Build/cpusupport.sh successfully
  * compiling Build/cpusupport-ARCH-FEATURE.c) then the C file containing the
  * corresponding run-time detection code (cpusupport_arch_feature.c) must be
  * compiled and linked in.
+ *
+ * There are a few features for which we do not have run-time checks:
+ * - X86_CPUID: compile-time is enough; if __get_cpuid() fails, then all the
+ *              x86 detection features will fail, but there's nothing we can
+ *              do about that.
+ * - X86_CUPID_COUNT: ditto.
```
Comment (cperciva) [inline cpusupport/cpusupport.h:129]: As much as I like the mental image of counting cupids, I don't think it's what you meant.

**Comment by cperciva** [comment]
LGTM once the typo is fixed.

**Comment by gperciva** [comment]
Rebased, ready for merge.

**Pull request merged**